src/data/validation/email.test.js

```diff
@@ -37,6 +38,45 @@ describe('email validation', () => {
       expect(validateEmailAddresses(validEmails)).toEqual(expectedResult);
     });
 
+    it('returns valid email addresses and indices when various unicode formatting is in the picture', () => {
+      const unicodeEmails = [
+        'dave.smith@\u200Fexample.com', // RLM after @
+        'john.doe@test.\u200Ecom', // LRM after '.' in domain
+        'alex\u200Ddoe@somewhere.com', // ZWJ between letters in local part
+        'sam\u200C@example.net', // ZWNJ at end of local part
+        '\u202Aalice@example.org', // LRE at start of address
+        'kate.jones@domain\u202B.com', // RLE before '.' in domain
+        'roger+\u202Csales@company.com', // PDF after '+' in local part
+        'chris@exa\u202Dmple.com', // LRO in domain name
+        'mary@contoso.com\u202E', // RLO at end of address
+        'anna\u2066.smith@mysite.com', // LRI before '.' in local part
+        'info@support.\u2067example.com', // RLI after subdomain '.'
+        'mark@example.c\u2068om', // FSI inside TLD
+        'tom.\u2069adams@example.com', // PDI after '.' in local part
+      ];
+
+      // Verify that the Unicode does exist in the email
+      const formatCharRegex = XRegExp('\\p{Cf}', 'g');
+      unicodeEmails.forEach((unicodeEmail) => {
+        const hasUnicodeFormat = XRegExp.test(unicodeEmail, formatCharRegex);
+        expect(hasUnicodeFormat).toBe(true);
+      });
+
+      const expectedResult = { ...baseResults };
+      expectedResult.validEmails = unicodeEmails.map(email => XRegExp.replace(email, formatCharRegex, '').trim());
```

Comment:
I just wanted to call out this general test pattern which is self-fulfilling. IMO, the code you are trying to test should be off-limits from being in the test.

In this case I'd define `unicodeEmails` above as a list of lists, one input and one expected output.

Reply:
Corrected with hard coded output and an individualized check at a char code level whether the unicode value exist 👍🏽